App/Dependencies/ExposureDetection/ImmuniExposureDetectionExecutor.swift

```diff
@@ -136,7 +136,136 @@ class ImmuniExposureDetectionExecutor: ExposureDetectionExecutor {
+      exposureInfoRiskScoreThreshold: Int,
+      userExplanationMessage: String,
+      enManager: ExposureNotificationManager,
+      tekProvider: TemporaryExposureKeyProvider,
+      now: @escaping () -> Date,
+      isUserCovidPositive: Bool,
+      forceRun: Bool,
+      exposureDetectionEU: [(String, Int?, Date?)]
+    ) -> Promise<ExposureDetectionOutcomeEU> {
+      return Promise(in: .custom(queue: Self.queue)) { resolve, _, _ in
+        guard #available(iOS 13.5, *) else {
+          // No exposure detection to perform, ever.
+          resolve(.noDetectionNecessary)
+          return
+        }
 
+        let timeSinceLastDetection = now().timeIntervalSince(lastExposureDetectionDate ?? .distantPast)
+        guard forceRun || timeSinceLastDetection >= exposureDetectionPeriod else {
+          // Exposure detection was performed recently
+          resolve(.noDetectionNecessary)
+          return
+        }
+
+        // Check for authorization
+        let status: ExposureNotificationStatus
+        do {
+          status = try await(enManager.getStatus())
+        } catch {
+          resolve(.error(.unableToRetrieveStatus(error)))
+          return
+        }
+
+        guard status.canPerformDetection else {
+          // No authorization
+          resolve(.error(.notAuthorized))
+          return
+        }
+        
+        var keyChunks: [TemporaryExposureKeyChunk] = []
+        
+        // tupla: (countryId,[TemporaryExposureKeyChunk])
+        var keyChunksMatrix: [(String, [TemporaryExposureKeyChunk])] = []
+
+        for exposureDetectionOfCountry in exposureDetectionEU {
+            
+            // Download the keys
+            let country:String = exposureDetectionOfCountry.0
+            var keyChunksOfCountry: [TemporaryExposureKeyChunk]
+            do {
+               keyChunksOfCountry = try await(
+                tekProvider
+                    .getLatestKeyChunksEU(latestKnownChunkIndex: exposureDetectionOfCountry.1, country: country)
+              )
+                keyChunks += keyChunksOfCountry
+                keyChunksMatrix.append((country, keyChunksOfCountry))
+            } catch {
+                continue
+            }
+        }
+       
+        defer {
+          // Cleanup the local files of the downloaded chunks
+          try? await(tekProvider.clearLocalResources(for: keyChunks))
+        }
+
+        guard !keyChunks.isEmpty else {
+          // No new keys. There is not detection to perform.
+          resolve(.noDetectionNecessary)
+          return
+        }
+
+//        let firstProcessedChunk = keyChunks.map { $0.index }.min()
+//          ?? AppLogger.fatalError("keyChunks cannot be empty at this stage")
+//
+//        let lastProcessedChunk = keyChunks.map { $0.index }.max()
+//          ?? AppLogger.fatalError("keyChunks cannot be empty at this stage")
+
+        var firstProcessedChunkEU:[(String, Int)] = []
+        var lastProcessedChunkEU:[(String, Int)] = []
```

Comment:
<!--
  0 failure: 
  1 warning:  swiftlint (0.39.2...
  
  
  DangerID: danger-id-Danger;
  File: App/Dependencies/ExposureDetection/ImmuniExposureDetectionExecutor.swift;
  Line: 222;
-->

- :warning: swiftlint (0.39.2): Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. (colon)